site/docs/concepts/architecture.md

```diff
@@ -12,15 +12,15 @@ The primary interaction object for a data user is the `FybrikApplication` custom
 
 Fybrik connects to external services to receive data governance decisions, metadata about datasets and credentials. Policies, assets and access credentials to the assets have to be defined before the user can run an application. The current abstraction supports 2 different [connectors](./connectors.md): one for data catalog and one for policy manager. It is designed in an open way so that multiple different catalog and policy frameworks of all kinds of cloud and on-prem systems can be supported. The data steward configures policies in an external policy manager over assets defined in an external data catalog. Dataset credentials are retrieved from Vault by using [Vault API](https://www.vaultproject.io/api). Vault uses a custom secret engine implemented with [HashiCorp Vault plugins system](./vault_plugins.md) to retrieve the credentials from where they are stored (data catalog for example).
 
 Once a developer submits a `FybrikApplication` CRD to Kubernetes, the FybrikApplicationController will make sure that all the specs are fulfilled and will make sure that the data is read/written/copied/deleted according to the data governance policies. The `FybrikApplication` holds metadata about the application such as the data assets required by the application, the processing purpose and the method of access the user wishes (protocol e.g. S3 or Arrow flight). 
-It uses this information to check with the data governance policy manager (4) if the data flow requested is allowed
-and whether restrictive policies such as masking or hashing have to be applied. It compiles plotters based on the governance decisions received via the data governance policy connector and chooses the modules (5) which are best fit for the requirements that the user specified regarding the access protocol and availability, and based on the [config policies](./config-policies.md) defined.  The plotter contains the modules required and all the information required to deploy them on the cluster where they will run, as well as the flow of data between the asset and the workload through the chosen modules.  
+The controller uses this information to check with the data-governance policy manager (4) if the data flow requested is allowed
+and whether restrictive policies such as masking or hashing have to be applied. It compiles a plotter based on the governance decisions received via the data governance policy connector and chooses the modules (5) which are best fit for the requirements that the user specified regarding the access protocol and availability, and based on the [config policies](./config-policies.md) defined. The plotter contains the modules required and all the information required to deploy them on the cluster where they will run, as well as the flow of data between the asset and the workload through the chosen modules. For the plotter to be optimal in terms of the defined optimization goals, the controller may use a [CSP-based optimizer](./optimizer.md).
```

Comment:
Continuing my comment from above, IT config policies can be hard policies indicating requirements or restrictions or they can indicate optimization goals.  A CSP engine (if activated) will ensure that the optimal solution for use of a dataset by a workload will be designated in the plotter.  If no CSP engine is installed soft policies will not be taken into account, and the manager will use the first (but not necessarily optimal) solution that meets all of the other requirements.